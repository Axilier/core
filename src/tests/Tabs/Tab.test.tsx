import { mount } from 'enzyme';
import React from 'react';
import { Tab } from '../../index';

const onClick = jest.fn();
const tabRow = mount(<Tab orientation={'row'}>click me</Tab>);
const tabColumn = mount(<Tab orientation={'column'}>click me</Tab>);
const tabUnDef = mount(<Tab>click me</Tab>);

describe('when rendered in an application', () => {
    describe('when the `onClick` prop is not set', () => {
        beforeAll(() => tabRow.simulate('click'));
        it('should cause nothing to happen', () => {
            expect(true).toBeTruthy();
        });
    });
    describe('when the `onClick` prop is set', () => {
        beforeAll(() => {
            tabRow.setProps({ onClick });
            tabRow.simulate('click');
        });
        it('should register each click and run the provided callback', () => {
            expect(onClick).toHaveBeenCalled();
        });
        afterAll(() => tabRow.setProps({ onClick }));
    });
    describe('when the `showSelectedShadow` is set to true', () => {
        beforeAll(() => tabRow.setProps({ showNotSelectedShadow: true }));
        it('should make the shadow div have a display property of block', () => {
            expect(
                tabRow.childAt(0).childAt(0).childAt(1).props().style,
            ).toHaveProperty('display', 'block');
        });
        afterAll(() => tabRow.setProps({ showNotSelectedShadow: undefined }));
    });
    describe('when the `showSelectedShadow` is set to false', () => {
        beforeAll(() => tabRow.setProps({ showNotSelectedShadow: false }));
        it('should make the shadow div have a display property of none', () => {
            expect(
                tabRow.childAt(0).childAt(0).childAt(1).props().style,
            ).toHaveProperty('display', 'none');
        });
        afterAll(() => tabRow.setProps({ showNotSelectedShadow: undefined }));
    });
});
describe('when the `orientation` is row', () => {
    it('should render correctly', () => {
        expect(tabRow).toMatchSnapshot();
    });
    it('should have the expected styles', () => {
        expect(tabRow.childAt(0).props().style).toEqual({
            paddingLeft: 'unset',
        });
    });
    it('should have the tabHorizontal className', () => {
        expect(tabRow.childAt(0).props().className).toEqual(
            'tab tabHorizontal',
        );
    });
});
describe('when the `orientation` is column', () => {
    it('should render correctly', () => {
        expect(tabColumn).toMatchSnapshot();
    });
    it('should have the expected styles', () => {
        expect(tabColumn.childAt(0).props().style).toEqual({
            paddingLeft: '20px',
        });
    });
    it('should only have the tab className', () => {
        expect(tabColumn.childAt(0).props().className).toEqual('tab ');
    });
});
describe('when the `orientation` is not defined', () => {
    it('should render correctly', () => expect(tabUnDef).toMatchSnapshot());
});
