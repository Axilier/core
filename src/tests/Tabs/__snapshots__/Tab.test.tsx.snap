// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the \`orientation\` is column should render correctly 1`] = `
<Tab
  orientation="column"
>
  <Layout
    center={true}
    className="tab "
    onClick={[Function]}
    orientation="column"
    style={
      Object {
        "paddingLeft": "20px",
      }
    }
  >
    <div
      className="tab "
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": "20px",
        }
      }
    >
      click me
      <div
        className="tabSelectedShadow verticalShadow"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
  </Layout>
</Tab>
`;

exports[`when the \`orientation\` is not defined should render correctly 1`] = `
<Tab>
  <Layout
    center={true}
    className="tab tabHorizontal"
    onClick={[Function]}
    orientation="row"
    style={
      Object {
        "paddingLeft": "unset",
      }
    }
  >
    <div
      className="tab tabHorizontal"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingLeft": "unset",
        }
      }
    >
      click me
      <div
        className="tabSelectedShadow horizontalShadow"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
  </Layout>
</Tab>
`;

exports[`when the \`orientation\` is row should render correctly 1`] = `
<Tab
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  orientation="row"
>
  <Layout
    center={true}
    className="tab tabHorizontal"
    onClick={[Function]}
    orientation="row"
    style={
      Object {
        "paddingLeft": "unset",
      }
    }
  >
    <div
      className="tab tabHorizontal"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingLeft": "unset",
        }
      }
    >
      click me
      <div
        className="tabSelectedShadow horizontalShadow"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
  </Layout>
</Tab>
`;
